net-test/bin/start.sh

```diff
@@ -0,0 +1,363 @@
+
+FAUCET_PORT=8080
+
+if [ -f "./config.sh" ]; then
+   source ./config.sh
+fi
+
+function exit_error() {
+   printf "$1" >&2
+   exit 1
+}
+
+function log() {
+   printf "%s" "$1" >&2
+}
+
+function logln() {
+   printf "%s\n" "$1" >&2
+}
+
+start_bitcoind() {
+   logln "Setting up Bitcoind..."
+   test -f "$BITCOIN_CONF".in || exit_error "No such file or directory: $BITCOIN_CONF"
+   mkdir -p "$BITCOIN_DATA_DIR" || exit_error "Failed to create Bitcoin data directory $BITCOIN_DATA_DIR"
+
+   log "  Generating Bitcoind config file..."
+   sed -r \
+     -e "s!@@BITCOIN_DATA_DIR@@!$BITCOIN_DATA_DIR!g" \
+     "$BITCOIN_CONF".in \
+     > "$BITCOIN_CONF"
+   logln "ok"
+   
+   log "Starting bitcoind..."
+   bitcoind -conf="$BITCOIN_CONF" >"$BITCOIN_LOGFILE" 2>&1 &
+   local BITCOIN_PID=$!
+   logln "PID $BITCOIN_PID"
+
+   while true; do 
+      bitcoin-cli -regtest -conf="$BITCOIN_CONF" ping >/dev/null 2>&1
+      if [ $? -eq 0 ]; then
+         break
+      fi
+      sleep 1
+   done
+
+   echo "$BITCOIN_PID"
+   return 0
+}
+
+start_bitcoind_controller() {
+   logln "Setting up Bitcoind controller..."
+   test -f "$BITCOIN_CONTROLLER_CONF".in || exit_error "No such file or directory: $BITCOIN_CONTROLLER_CONF.in"
+
+   log "  Generating Bitcoind controller config file..."
+   local NOW="$(date +%s)"
+   sed -r \
+    -e "s/@@BITCOIN_NEON_CONTROLLER_GENESIS_TIMESTAMP@@/$NOW/g" \
+    "$BITCOIN_CONTROLLER_CONF".in \
+    > "$BITCOIN_CONTROLLER_CONF"
+   logln "ok"
+
+   log "Starting bitcoind controller..."
+   bitcoin-neon-controller "$BITCOIN_CONTROLLER_CONF" >"$BITCOIN_NEON_CONTROLLER_LOGFILE" 2>&1 &
+   local RC=$?
+   local BITCOIN_NEON_CONTROLLER_PID=$!
+
+   if [ $RC -ne 0 ]; then 
+      logln "FAILED"
+      return 1
+   fi
+
+   logln "PID $BITCOIN_NEON_CONTROLLER_PID"
+
+   echo "$BITCOIN_NEON_CONTROLLER_PID"
+   return 0
+}
+
+start_faucet() {
+   logln "Setting up Bitcoin faucet..."
+   test -f "$BITCOIN_CONTROLLER_CONF" || exit_error "No such file or directory: $BITCOIN_CONTROLLER_CONF"
+   test -f "$BITCOIN_CONF" || exit_error "No such file or directory: $BITCOIN_CONF"
+
+   local PRIVKEY="$(cat "$BITCOIN_CONTROLLER_CONF" | grep "# WIF" | cut -d ' ' -f 3)"
+
+   log "  Importing faucet private key..."
+   bitcoin-cli -conf="$BITCOIN_CONF" importprivkey "$PRIVKEY" >"$FAUCET_LOGFILE" 2>&1
+   local RC=$?
+   if [ $RC -ne 0 ]; then
+      logln "FAILED!"
+      return 1
+   fi
+   logln "ok"
+
+   local MINER_ADDR="$(cat "$BITCOIN_CONTROLLER_CONF" | grep "miner_address =" | cut -d ' ' -f 3 | sed -r 's/"//g')"
+
+   log "  Importing miner address $MINER_ADDR..."
+   
+   echo "Try importing miner address $MINER_ADDR" >>"$FAUCET_LOGFILE"
+   bitcoin-cli -conf="$BITCOIN_CONF" importaddress "$MINER_ADDR" >>"$FAUCET_LOGFILE" 2>&1
+   RC=$?
+   if [ $RC -ne 0 ]; then 
+      logln "FAILED!"
+      return 1
+   fi
+
+   logln "ok"
+
+   log "Starting Bitcoin faucet..."
+   faucet.sh "$FAUCET_PORT" "$BITCOIN_CONF" >"$FAUCET_LOGFILE" 2>&1 &
```

Comment:
using `./faucet.sh` everywhere would allow one to run from the bin directory without having to add it to the PATH

Reply:
This set of programs is ultimately meant to run in a Docker container, so `faucet.sh` will be guaranteed to be in the `$PATH`.